### DISCUSSION TITLE
Restore from Instance Backup

### DISCUSSION THREAD

**Discussion by calvinalx** [coollabsio/coolify#1684] [Feature Requests]
I noticed that we have **Instance backup** at Settings page, and was able to download Coolify database dump
e.g. `pg-dump-coolify-1706522852.dmp`

How do we restore this backup to a new Coolify install? I can't find this anywhere on the docs.
Assuming this is just another Postgres container running, is it possible just import it into the database of a new Coolify setup and possibly re-generate all the configs/file?

Thanks!

**Comment by TomEbeyer** [comment]
Same question - How to restore coolify backup from the .dmp?

**Comment by pedrosorrentino** [comment]
Another one with the doubt of how to restore the backup. Any idea?

**Comment by xyba1337** [comment]
@andrasbacsai Any help, please??

**Comment by HarleySalas** [comment]
I also really need to move my Coolify over to a new VPS. I've so far wasted a day, trying to figure out how to get it to work.

**Comment by devdjdjdj** [comment]
Coolify runs a postgres db underneath. try this [pg_dump](url)

You can find the credentials in your settings http://youip:8000/settings#backup

<img width="1312" alt="Screenshot 2024-06-09 at 7 39 24 PM" src="https://github.com/coollabsio/coolify/assets/8934007/d08a0b4d-710a-4dde-9224-d8434e46c609">

**Reply by enesbala5** [reply]
Having created a db dumb, what is the command to import it and overwrite the current configuration.

I understand it probably is pg_restore, however I would really appreciate some help on the exact command. Thank you

**Reply by devdjdjdj** [reply]
pg_restore -U username -d dbname -1 filename.dump

**Comment by enesbala5** [comment]
I have managed to run the pg_restore command within the docker container 'coolify-db'. However these are NOT working to update the db successfully, even though the operation is running without issues.
This is the exact chain of commands I am running:

docker exec -it coolify-db 'bash'

-> I uploaded an image to filebin.net and am running:
wget -O db.dmp <download_link>

pg_restore --verbose --clean -h localhost -U coolify -d postgres db.dmp

---

After this step you can see all the fields updating, however no changes in the dashboard.. Don't know where to go from here.
I have tried setting the db to 'coolify-db' as it's displayed on the instance-backup but to no avail. Also password authentication doesn't seem to be required.

**Comment by mohammad-malik** [comment]
Would love to know the answer to this?

**Reply by enesbala5** [reply]
@andrasbacsai This would be really helpful - backup and restore is a very important feature of Coolify - and a massive hassle to have to do manually.

**Comment by Shreyaan** [comment]
https://gist.github.com/Geczy/83c1c77389be94ed4709fc283a0d7e23

something like this will work

**Comment by heyJordanParker** [comment]
Any updates here?

Even the backup docs page is not working.

**Reply by devdjdjdj** [reply]
Looks like the astro blog is broken.

You can find the missing docs here - https://github.com/coollabsio/documentation-coolify/blob/main/src/content/docs/knowledge-base/databases/backups.mdx

**Comment by hnykda** [comment]
Thanks for the hints. I find this pretty bad that coolify doesn't support this officially. I was trying to recover from the autobackup, but was just getting weird 500 errors such as:

```
Error: The MAC is invalid. (View: /var/www/html/resources/views/livewire/project/resource/index.blade.php)
```

even when using the same `.env` file. But there is zero documentation on that aspect (I was really guessing db names etc.).

**Comment by renatoaraujoc** [comment]
Hello all,

The best case scenario for a complete coolify backup would be:

Backup script:

1. Stop all containers related to coolify and coolify itself
2. Backup everything under /data/coolify religiously
3. Backup every single volume created by coolify and its apps
4. Create a coolify_{{date}}.tar.gz with everything inside

I believe that if you backup the coolify-db's volume, you don't need to mess with pg_dump or pg_restore

Restore script:

1. Take an user input which you'd type the generated backup from the backup script, ie: colify_{{date}}.tar.gz
2. Restore /data/coolify
3. Restore the coolify's ssh
4. Restore all docker volumes
5. Optimally start all apps or not

Obviously there's should be more steps into this.

The backup and restore script could be baked inside the coolify's UI itself, so you back it up from there and on a new server (or even the same server (older state)) you restore from there as well.

The biggest issue here is to separate the concerns on coolify's data itself and the apps' data, it's not an easy solution to bake. If you choose to to back to an old coolify state, what happens to an app that has new data? Lots of stuff has to be thought of.

@andrasbacsai Coolify looks enough mature today, this functionality (imo) is very needed.

**Reply by dwillitzer** [reply]
## Coolify Database Restore Tool

Hi Coolify community,

I wanted to share a utility script I developed called **Coolify Database Restore Tool**. This tool is designed specifically for managing and restoring PostgreSQL database backups in Coolify environments.  

### 🔹 Purpose  
The script automates the process of restoring a Coolify PostgreSQL database backup. It lists available backups (displaying human-readable dates), extracts the selected backup, copies it to the Coolify database container, and then executes the necessary commands to:  
- Terminate active connections  
- Drop and recreate the database  
- Restore the backup  
- Restart Coolify containers to ensure the environment is ready  

### 🚀 Features  
✅ Lists available backups with clear, human-readable dates  
✅ Handles backup extraction using `gunzip`  
✅ Copies the backup file into the Coolify database container  
✅ Executes database commands inside the container for restoration  
✅ Restarts Coolify containers post-restore  

### 📋 Prerequisites  
- Docker must be installed and running  
- Access to the Coolify backup directory with proper permissions to:  
  - Read backup files  
  - Execute Docker commands  
  - Restart containers  

### ⚙️ Configuration  
The script assumes default Coolify container and database naming conventions. You can adjust the configuration variables in the script if your setup differs:  
```bash
BACKUP_DIR="/data/coolify/backups/coolify/coolify-db-hostdockerinternal"
CONTAINER_NAME="coolify-db"
DB_USER="coolify"
DB_NAME="coolify"
```

⚠️ Disclaimer

This tool is a community script and is not officially affiliated with Coolify. It performs destructive operations (dropping and recreating databases), so please test in a non-production environment first and ensure you have proper backups before using it.

For full details—including usage examples and troubleshooting tips—please refer to the repository:

🔗 [GitHub Repository: dwillitzer/coolify-db-restore](https://github.com/dwillitzer/coolify-db-restore)

I’m looking forward to your feedback and any suggestions for improvements!

Cheers,
Daniel Willitzer

**Reply by LunaVoid** [reply]
> Hello all,
> 
> The best case scenario for a complete coolify backup would be:
> 
> Backup script:
> 
>     1. Stop all containers related to coolify and coolify itself
> 
>     2. Backup everything under /data/coolify religiously
> 
>     3. Backup every single volume created by coolify and its apps
> 
>     4. Create a coolify_{{date}}.tar.gz with everything inside
> 
> 
> I believe that if you backup the coolify-db's volume, you don't need to mess with pg_dump or pg_restore
> 
> Restore script:
> 
>     1. Take an user input which you'd type the generated backup from the backup script, ie: colify_{{date}}.tar.gz
> 
>     2. Restore /data/coolify
> 
>     3. Restore the coolify's ssh
> 
>     4. Restore all docker volumes
> 
>     5. Optimally start all apps or not
> 
> 
> Obviously there's should be more steps into this.
> 
> The backup and restore script could be baked inside the coolify's UI itself, so you back it up from there and on a new server (or even the same server (older state)) you restore from there as well.
> 
> The biggest issue here is to separate the concerns on coolify's data itself and the apps' data, it's not an easy solution to bake. If you choose to to back to an old coolify state, what happens to an app that has new data? Lots of stuff has to be thought of.
> 
> @andrasbacsai Coolify looks enough mature today, this functionality (imo) is very needed.

This is a great point, ran into a couple of issues that would be completely and utterly solved by this. Do you know if only settings for coolify are stored in the database or also the docker compose and other env variables are. Because than the current restore could spin up the containers again when restored and you'd be in a similar state. Though internal state would be different for some containers.  I'll need to poke around more in the database and learn how coolify works, but I was curious if there was any more motion on this topic?

**Comment by RatulMaharaj** [comment]
This section of the docs might be helpful for those looking to restore backups: https://coolify.io/docs/databases/backups#restore-command

**Reply by enesbala5** [reply]
I have done a `pg_restore` on Coolify before, and it has done nothing. Only the tools which are mentioned in this issue have managed to successfully restore my data.

Regardless I will try again. Thanks

**Reply by RatulMaharaj** [reply]
I have the box to backup all database checked - that might make a difference. I think it results in a `.gz` backup file, which I am then to restore with the UI.

**Comment by davidkarolyi** [comment]
Here is an other simple way to restore a backup (on Mac):
- Make sure your db instance is publicly available through a proxy (you can configure this under your Database resource)
- Download the backup you wanna restore from the dashboard under your database, under the Backups tab.
- Install `pg_restore` if you haven't already:
  ```sh
  brew install libpq
  export PATH="/opt/homebrew/opt/libpq/bin:$PATH"
  pg_restore --version
  ```
- Restore the backup from your local copy
  ```sh
  PGPASSWORD='password-of-your-public-db' \
  pg_restore --verbose --clean --no-acl --no-owner \
  --host=<ip-address-of-your-machine> --port=5432 \
  --username=postgres --dbname=postgres \
  ~/Downloads/<backup-name>.dmp
  ```
